aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -66,377 +98,362 @@ static std::vector<int64_t> conv_output_size(
+    IntArrayRef padding, IntArrayRef output_padding, IntArrayRef stride,
+    IntArrayRef dilation, int64_t groups, bool bias_defined) {
+
+  auto grad_weight = at::empty(weight_size, grad_output.options());
+  Tensor grad_bias;
   if (bias_defined) {
-    grad_bias_memory.reset(new memory({{{bias_tz}, data_t, format_x}, cpu_engine},
-      grad_bias.data_ptr()));
-    conv_backward_weight.reset(new convolution_backward_weights(*conv_backward_weight_pd,
-      input_memory, grad_output_memory, grad_weight_memory, *grad_bias_memory));
-  } else {
-    conv_backward_weight.reset(new convolution_backward_weights(*conv_backward_weight_pd,
-      input_memory, grad_output_memory, grad_weight_memory));
+    grad_bias = at::empty({grad_output.size(1)}, grad_output.options());
   }
 
-  net.push_back(*conv_backward_weight);
+  ConvolutionArgs args(input, grad_weight, grad_output, padding,
+      output_padding, stride, dilation, groups, bias_defined, true);
+
+  auto dtype = get_mkldnn_dtype(input);
+  desc x_desc(args.input_tz, dtype);
+  desc grady_desc(args.output_tz, dtype);
+
+  itensor x_, grady_, gradw_, gradw/*user layout*/, gradb_, gradb/*user layout*/;
+  x_.init(x_desc, input.data_ptr());
+  grady_.init(grady_desc, grad_output.data_ptr());
+
+  desc gradw_desc(args.weight_tz, grady_.get_data_type());
+  gradw.init(gradw_desc, grad_weight.data_ptr());
 
-  if (grad_weight_memory != grad_weight_usr_memory) {
-    net.push_back(reorder(grad_weight_memory, grad_weight_usr_memory));
+  if (bias_defined) {
+    desc gradb_desc(args.bias_tz, grady_.get_data_type());
+    gradb.init(gradb_desc, grad_bias.data_ptr());
+    convolution_transpose_backward_weights::compute(x_, grady_, args.weight_tz, gradw_, gradb_,
+        args._stride, args._padding, args._padding_r);
+    reorder::compute(gradb_, gradb);
+  } else {
+    convolution_transpose_backward_weights::compute(x_, grady_, args.weight_tz, gradw_,
+        args._stride, args._padding, args._padding_r);
   }
 
-  Stream::Instance().get_stream().submit(net);
+  reorder::compute(gradw_, gradw);
+  auto grad_weight_t = grad_weight.transpose(0 , 1);
 
-  return std::tuple<at::Tensor, at::Tensor>{grad_weight, grad_bias};
+  return std::tuple<Tensor, Tensor>{grad_weight_t, grad_bias};
 }
 
-std::tuple<at::Tensor,at::Tensor,at::Tensor> mkldnn_convolution_backward(
-    const at::Tensor& input, const at::Tensor& grad_output_t, const at::Tensor& weight,
-    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups, std::array<bool,3> output_mask)
-{
+std::tuple<Tensor, Tensor, Tensor> mkldnn_convolution_transpose_backward(
+    const Tensor& input, const Tensor& grad_output_t, const Tensor& weight_t,
+    IntArrayRef padding, IntArrayRef output_padding, IntArrayRef stride,
+    IntArrayRef dilation, int64_t groups, std::array<bool,3> output_mask) {
+
   Tensor grad_output = grad_output_t.contiguous();
+  auto weight = weight_t.transpose(0,1).contiguous();
```

Comment:
Ick. Really?!

Reply:
this is because PyTorch weight format for transposed conv is `{iohw}`: input_channel, output_channel, kernel_height, kernel_weight.
but `mkldnn` weight format for transposed conv is `oihw` so we need a transpose on weight here.
anyway, for later release of `mkldnn`, `iohw` is supported. which means this won't be needed by then.